packages/http-server/src/http-server.ts

```diff
@@ -0,0 +1,85 @@
+// Copyright IBM Corp. 2017,2018. All Rights Reserved.
+// Node module: @loopback/http-server
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {createServer, Server} from 'http';
+import {RestServer, HttpRequestListener} from '@loopback/rest';
+import {AddressInfo} from 'net';
+import * as pEvent from 'p-event';
+
+/**
+ * Object for specifyig the HTTP / HTTPS server options
+ */
+export type HttpOptions = {
+  port: number;
+  host: string | undefined;
+};
+
+/**
+ * HTTP / HTTPS server used by LoopBack's RestServer
+ *
+ * @export
+ * @class HttpServer
+ */
+export class HttpServer {
+  private restServer: RestServer;
+  private httpPort: number;
+  private httpHost: string | undefined;
+  private httpServer: Server;
+
+  /**
+   * @param restServer
+   * @param httpOptions
+   * @param httpRequestListener
+   */
+  constructor(
+    restServer: RestServer,
+    httpOptions: HttpOptions,
+    httpRequestListener: HttpRequestListener,
+  ) {
+    this.restServer = restServer;
+    this.httpPort = httpOptions.port;
+    this.httpHost = httpOptions.host;
+    this.httpServer = createServer(httpRequestListener);
+  }
+
+  /**
+   * Starts the HTTP / HTTPS server
+   */
+  public start(): Promise<void> {
+    this.httpServer.listen(this.httpPort, this.httpHost);
+    return new Promise<void>(async (resolve, reject) => {
+      try {
+        await pEvent(this.httpServer, 'listening');
+        const address = this.httpServer.address() as AddressInfo;
+        this.restServer.bind('rest.port').to(address.port);
```

Comment:
@bajtos  I wanted to use `RestBindings.PORT`, but `RestBindings` itself is undefined when `{RestBindings}` is imported into `http-sever.ts` - `http-sever.ts` is executed before the `RestBindings` object is created.

If there is a solution, do let me know.

Reply:
IMO, `HttpServer` should not deal with dependency injection at all. Instead, it should provide properties like `port`, `hostname` and perhaps `url` that can be used by consumers to update any bindings as necessary.

```ts
class HttpServer {
  get port(): number {
    return this.address().port;
  }

  address: AddressInfo {
    return this.address() as AddressInfo;
  }
}
```

Later in RestServer:

```js
async start() {  
  // create http server and start it, then:
  this.bind(RestBindings.PORT).to(server.port);
}
```